From e7a7189133df950438f5298c9cc18a70d2a40769 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Tue, 20 Jan 2015 16:02:44 +0800
Subject: [PATCH] MLK-10117-4: mxsfb: Add lcdif register config in enable
 panel function

Lcdif register configuration will lost in dispmix power off,
so register should been configured again when panel power up.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/video/mxsfb.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/video/mxsfb.c b/drivers/video/mxsfb.c
index f7fae186..50e84b4 100644
--- a/drivers/video/mxsfb.c
+++ b/drivers/video/mxsfb.c
@@ -4,7 +4,7 @@
  * This code is based on:
  * Author: Vitaly Wool <vital@embeddedalley.com>
  *
- * Copyright 2008-2014 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2008-2015 Freescale Semiconductor, Inc. All Rights Reserved.
  * Copyright 2008 Embedded Alley Solutions, Inc All Rights Reserved.
  *
  * This program is free software; you can redistribute it and/or
@@ -239,6 +239,7 @@ static const struct mxsfb_devdata mxsfb_devdata[] = {
 
 static int mxsfb_map_videomem(struct fb_info *info);
 static int mxsfb_unmap_videomem(struct fb_info *info);
+static int mxsfb_set_par(struct fb_info *fb_info);
 
 /* enable lcdif axi clock */
 static inline void clk_enable_axi(struct mxsfb_info *host)
@@ -515,6 +516,9 @@ static void mxsfb_enable_controller(struct fb_info *fb_info)
 	/* Clean soft reset and clock gate bit if it was enabled  */
 	writel(CTRL_SFTRST | CTRL_CLKGATE, host->base + LCDC_CTRL + REG_CLR);
 
+	/* reconfigure the lcdif after */
+	mxsfb_set_par(&host->fb_info);
+
 	writel(CTRL2_OUTSTANDING_REQS__REQ_16,
 		host->base + LCDC_V4_CTRL2 + REG_SET);
 
-- 
1.8.0

